### PR TITLE
add k8s-1.30 package and variants boilerplate

### PR DESCRIPTION
**Pull request by ginglis13**
**Issue number:**

Closes #

**Description of changes:**

Adds boiler plate for k8s-1.30 variants based on previous addition of k8s variants. This commit does not add the actual k8s-1.30 sources, but rather sets the stage for incremental changes to k8s-1.30 variants.

This adds a k8s-1.30 package. Existing 1.29 models are moved to be 1.30, with symlinks created for 1.29 models to point to the new locations. For now, the k8s-1.30 package points to the k8s-1.29 source.

**Testing done:**

No testing other than `cargo make -e BUILDSYS_VARIANT=aws-k8s-1.30`, just boilerplate. PR to add real k8s-1.30 sources will include full testing.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: packages/ecr-credential-provider-1.30/ecr-credential-provider-1.30.spec [hunk 1]
```diff
@@ -0,0 +1,46 @@
+%global goproject github.com/kubernetes
+%global gorepo cloud-provider-aws
+%global goimport %{goproject}/%{gorepo}
+
+%global gover 1.29.0
+# %global gover 1.30.0
```
Comment (bcressey) [inline packages/ecr-credential-provider-1.30/ecr-credential-provider-1.30.spec]: nit: usually for macros it's not enough to comment them, you also have to escape them:
```suggestion
# %%global gover 1.30.0
```

**Review by bcressey — APPROVED** [review]
LGTM. I'd be happy to see this merged and get the boring part out of the way early 😀

**Comment by ginglis13** [comment]
^escape commented out macros

**Review by bcressey — APPROVED** [review]

**Review by yeazelm — APPROVED** [review]

**Comment by ginglis13** [comment]
^fix build by specifying kubelet dependency on ecr-credential-provider-1.29, since ecr-credential-provider-1.30 isn't available yet.

error was:
```
nothing provides bottlerocket-ecr-credential-provider-1.30 needed by bottlerocket-kubelet-1.30-1.29.1-1.aarch64 from repo
```

built all k8s-1.30 variants to confirm.

**Review by bcressey — APPROVED** [review]

**Pull request merged**